### ISSUE TITLE
GPU Support

### ISSUE THREAD

**Issue by loretoparisi** [criteo/autofaiss#185]
I see there is a `--use_gpu ` parameter in the `build_index`, but the requirements are for `faiss-cpu`.
Is it possible instead to use [faiss-gpu](https://anaconda.org/pytorch/faiss-gpu) package?

**Comment by rom1504** [comment]
Do you have an use case ?

GPUs do not speed up faiss in almost all cases

**Comment by loretoparisi** [comment]
I'm running a distributed training of a embedding model (dim. 256 or 512) on a large dataset (10^6) using torch lightning. Training runs in GPU, building the k-NN index is cpu and I wanted to benchmark gpu performances of faiss. Agreed that speed here may not be significative.

**Comment by rom1504** [comment]
1M embeddings of dimensions 512 is a very small dataset. You can do this on cpu on a small machine in less than one hour

Autofaiss is being used often on 5B embeddings of dimensions 768. It runs in a few hours if using 10 nodes

The only use case where GPU might be useful is with very large amount of centroids of the IVF. Above 1M centroids. That's only helpful if you need a very high recall for > 100M points

**Comment by rom1504** [comment]
All that said you can indeed install faiss GPU instead of faiss cpu

**Comment by loretoparisi** [comment]
thanks a lot for the details!

**Comment by paulthemagno** [comment]
After installing `faiss-gpu`, I'm getting error on `import faiss` as well

**Comment by rom1504** [comment]
see https://github.com/kyamagu/faiss-wheels/issues/77#issuecomment-1977901285
